src/main/java/com/arpnetworking/kairosdb/aggregators/MovingWindowAggregator.java

```diff
@@ -0,0 +1,294 @@
+package com.arpnetworking.kairosdb.aggregators;
+
+import org.joda.time.Chronology;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeField;
+import org.joda.time.DateTimeZone;
+import org.joda.time.chrono.GregorianChronology;
+import org.json.JSONException;
+import org.json.JSONWriter;
+import org.kairosdb.core.DataPoint;
+import org.kairosdb.core.aggregator.AggregatedDataPointGroupWrapper;
+import org.kairosdb.core.annotation.FeatureComponent;
+import org.kairosdb.core.datastore.DataPointGroup;
+import org.kairosdb.core.datastore.TimeUnit;
+import org.kairosdb.core.aggregator.RangeAggregator;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+@FeatureComponent(name = "movingWindow", description = "Creates a moving window of datapoints")
+public class MovingWindowAggregator extends RangeAggregator {
+    protected long m_startTime = 0L;
```

Comment:
Looks like this is using arp networking pp and its associated checkstyle. You will need to rename this to just `_` prefix and move them after the methods.